base/src/Data/Macaw/Memory.hs

```diff
@@ -465,6 +465,20 @@ instance MemWidth w => Integral (MemInt w) where
     where (q,r) = memIntValue x `quotRem` memIntValue y
   toInteger = toInteger . memIntValue
 
+instance MemWidth w => Bits (MemInt w) where
+  MemInt x .&. MemInt y = memInt (x .&. y)
+  MemInt x .|. MemInt y = memInt (x .|. y)
+  MemInt x `xor` MemInt y = memInt (x `xor` y)
+  complement (MemInt x) = memInt (complement x)
+  MemInt x `shift` i = memInt (x `shift` i)
+  MemInt x `rotate` i = MemInt (x `rotate` i)
```

Comment:
Should this use the `memInt` smart constructor for consistency with the other definitions?

Reply:
Indeed that'd be better. Will fix.